static/js/sefaria/sefaria.js

```diff
@@ -2206,7 +2206,20 @@ _media: {},
       let refToCheck = Array.isArray(ref) ? ref[0] : ref;
       const parsedRef = Sefaria.parseRef(refToCheck);
       const book = Sefaria.index(parsedRef.index);
-      return this.isCommentaryWithBaseText(book);
+      if (!this.isCommentaryWithBaseText(book)) {
+          return false;
+      }
+
+      // by this point, we know the book is in the right form, but we still need to check that the ref is in the right form
+      // The ref 'Ramban on Genesis, Introduction 1' shouldn't generate "Genesis, Introduction 1"
+      // this can be tested by modifying the ref and then calling parseRef on the modified ref
+      const parsedRefCopy = Object.create(parsedRef);
+      const baseText = book.base_text_titles[0];
+      parsedRefCopy.ref = parsedRefCopy.ref.replace(book.title, baseText);
+      if (Sefaria.parseRef(parsedRefCopy.ref).error) {
```

Comment:
this could probably be:
`return !!Sefaria.parseRef(parsedRefCopy.ref).error`
If I understand this correctly.